e2e/start_work_test.go

```diff
@@ -0,0 +1,119 @@
+	_ "github.com/ozontech/file.d/plugin/action/flatten"
+	_ "github.com/ozontech/file.d/plugin/action/join"
+	_ "github.com/ozontech/file.d/plugin/action/join_template"
+	_ "github.com/ozontech/file.d/plugin/action/json_decode"
+	_ "github.com/ozontech/file.d/plugin/action/json_encode"
+	_ "github.com/ozontech/file.d/plugin/action/keep_fields"
+	_ "github.com/ozontech/file.d/plugin/action/mask"
+	_ "github.com/ozontech/file.d/plugin/action/modify"
+	_ "github.com/ozontech/file.d/plugin/action/parse_es"
+	_ "github.com/ozontech/file.d/plugin/action/parse_re2"
+	_ "github.com/ozontech/file.d/plugin/action/remove_fields"
+	_ "github.com/ozontech/file.d/plugin/action/rename"
+	_ "github.com/ozontech/file.d/plugin/action/set_time"
+	_ "github.com/ozontech/file.d/plugin/action/throttle"
+	_ "github.com/ozontech/file.d/plugin/input/dmesg"
+	_ "github.com/ozontech/file.d/plugin/input/fake"
+	_ "github.com/ozontech/file.d/plugin/input/file"
+	_ "github.com/ozontech/file.d/plugin/input/http"
+	_ "github.com/ozontech/file.d/plugin/input/journalctl"
+	_ "github.com/ozontech/file.d/plugin/input/k8s"
+	_ "github.com/ozontech/file.d/plugin/input/kafka"
+	_ "github.com/ozontech/file.d/plugin/output/devnull"
+	_ "github.com/ozontech/file.d/plugin/output/elasticsearch"
+	_ "github.com/ozontech/file.d/plugin/output/file"
+	_ "github.com/ozontech/file.d/plugin/output/gelf"
+	_ "github.com/ozontech/file.d/plugin/output/kafka"
+	_ "github.com/ozontech/file.d/plugin/output/postgres"
+	_ "github.com/ozontech/file.d/plugin/output/s3"
+	_ "github.com/ozontech/file.d/plugin/output/splunk"
+	_ "github.com/ozontech/file.d/plugin/output/stdout"
+)
+
+// e2eTest is the general interface for e2e tests
+// Configure prepares config for your test
+// Send sends message in pipeline and waits for the end of processing
+// Validate validates result of the work
+type e2eTest interface {
+	Configure(t *testing.T, conf *cfg.Config, pipelineName string)
+	Send(t *testing.T)
+	Validate(t *testing.T)
+}
+
+func startForTest(t *testing.T, e e2eTest, configPath, pipelineName string) *fd.FileD {
+	conf := cfg.NewConfigFromFile(configPath)
+	e.Configure(t, conf, pipelineName)
+	filed := fd.New(conf, "off")
+	filed.Start()
+	return filed
+}
+
+func TestE2EStabilityWorkCase(t *testing.T) {
+	testsList := []struct {
+		e2eTest
+		cfgPath      string
+		pipelineName string
+	}{
+		{
+			e2eTest: &file_file.Config{
+				Count:   10,
+				Lines:   500,
+				RetTime: "1s",
+			},
+			cfgPath:      "./file_file/config.yml",
+			pipelineName: "test_file_file",
+		},
+		{
+			e2eTest: &http_file.Config{
+				Count:   10,
+				Lines:   500,
+				RetTime: "1s",
+			},
+			cfgPath:      "./http_file/config.yml",
+			pipelineName: "test_http_file",
+		},
+		{
+			e2eTest: &kafka_file.Config{
+				Topic:     "quickstart5",
+				Broker:    "localhost:9092",
+				Count:     500,
+				RetTime:   "1s",
+				Partition: 4,
+			},
+			cfgPath:      "./kafka_file/config.yml",
+			pipelineName: "test_kafka_file",
```

Comment:
Why you need pipeline name here? You already defined a pipeline in a config file, just use it.